sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1130,6 +1130,7 @@ static bool _resolve_addr(const ipv6_addr_t *dst, gnrc_netif_t *netif,
             entry = _nib_nc_add(dst, (netif != NULL) ? netif->pid : 0,
                                 GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE);
             if (entry == NULL) {
+                gnrc_pktbuf_release(pkt);
```

Comment:
Hm, while releasing the pktsnip in case of ENOMEM is sensible, might this not interfere with the lines in https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/nib/nib.c#L203:L209, for `netif == NULL`? From looking at the code, this line can also be reached when `netif == NULL`, in which case the pktsnip is released and later again with `gnrc_pktbuf_release_error`.

Reply:
For that case we don't land here. Due to lazy evaluation `netif == NULL` is `true` and we go directly into

https://github.com/RIOT-OS/RIOT/blob/f8a7c026f2a83a29323bfcef19b88f8cd28a0d47/sys/net/gnrc/network_layer/ipv6/nib/nib.c#L206-L210

Reply:
(so `_resolve_addr()` isn't called in the first place)

Reply:
Okay, I see. I do not understand the intention of having checks for `netif != NULL` in several places of `_resolve_addr()`, though. But that's another matter.

Reply:
Thats for other calls, not the one you were referring to.

Reply:
Okay code-wise ACK, I will give the script a shot now.